File: src/LC442_FindAllDuplicates/LC442.java
package LC442_FindAllDuplicates;

import java.util.ArrayList;
import java.util.List;

public class LC442 {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> list = new ArrayList<>();

        for(int i=0;i<nums.length;i++){

            int index = Math.abs(nums[i])-1;

            if(nums[index]<0)
                list.add(index+1);
            nums[index] = -nums[index];
        }

        return list;
    }
}

File: src/LC442_FindAllDuplicates/LC442.cpp
class LC442 {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        int n = nums.size();
        for(int num : nums)
        {
            int temp = num;
            while(temp > n) temp -= n;
            int index = temp-1;
            nums[index] += n;
        }
        vector<int> ans;
        for(int i=0; i<n; i++)
            if(nums[i] > 2*n)
                ans.push_back(i+1);
        return ans;
    }
};